a10_octavia/network/drivers/neutron/a10_octavia_neutron.py

```diff
@@ -122,6 +122,96 @@ def unplug_trunk_subports(self, trunk_id, subports):
+
+    def _remove_security_group(self, port, sec_grp_id):
+        port['security_groups'].remove(sec_grp_id)
+        payload = {'port': {'security_groups': port['security_groups']}}
+        try:
+            self.neutron_client.update_port(port['id'], payload)
+        except neutron_client_exceptions.PortNotFoundClient as e:
+            raise base.PortNotFound(str(e))
+        except Exception as e:
+            raise base.NetworkException(str(e))
+
+    def _cleanup_port(self, vip_port_id, port):
+        try:
+            self.neutron_client.delete_port(port['id'])
+        except (neutron_client_exceptions.NotFound,
+                neutron_client_exceptions.PortNotFoundClient):
+            if port['id'] == vip_port_id:
+                LOG.debug('VIP port %s already deleted. Skipping.', port['id'])
+            else:
+                LOG.warning("Can't deallocate instance port {0} because it "
+                            "cannot be found in neutron. "
+                            "Continuing cleanup.".format(port['id']))
+        except Exception:
+            message = _('Error deleting VIP port_id {port_id} from '
+                        'neutron').format(port_id=port['id'])
+            LOG.exception(message)
+            raise base.DeallocateVIPException(message)
+
+    def _get_instance_ports_by_subnet(self, compute_id, subnet_id):
+        ports = []
+        amp_ports = self.neutron_client.list_ports(device_id=compute_id)
+        for port in amp_ports.get('ports', []):
+            for fixed_ips in port.get('fixed_ips', []):
+                if fixed_ips.get('subnet_id') == subnet_id:
+                    ports.append(port)
+        return ports
+
+    def _get_ports_by_security_group(self, sec_grp_id):
+
+        # Handle backwards compatibility issue with neutronclient
+        try:
+            all_ports = self.neutron_client.list_ports(project_id=self.project_id)
+        except neutron_client_exceptions.BadRequest:
+            all_ports = self.neutron_client.list_ports(project=self.project_id)
+
+        filtered_ports = []
+        for port in all_ports.get('ports', []):
+            if sec_grp_id in port.get('security_groups', []):
+                filtered_ports.append(port)
+        return filtered_ports
+
+    def deallocate_vip(self, loadbalancer, lb_count_subnet):
+        """Delete the vrrp_port (instance port) in case nova didn't
+        This can happen if a failover has occurred.
+        """
+        ports = []
+        sec_grp = None
+        vip_port_id = loadbalancer.vip.port_id
+        if self.sec_grp_enabled:
+            sec_grp = self._get_lb_security_group(loadbalancer.id)
+            if sec_grp:
+                ports = self._get_ports_by_security_group(sec_grp['id'])
+        if self.sec_grp_enabled or not ports:
+            for amphora in loadbalancer.amphorae:
+                self._get_instance_ports_by_subnet(amphora.compute_id, loadbalancer.vip.subnet_id)
```

Comment:
Are we trying to get instance ports if didn't get ports by security group? like:
port = self._get_instance_ports_by_subnet(amphora.compute_id, loadbalancer.vip.subnet_id)